bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -3008,6 +2976,251 @@ private static boolean containsType(Set<Object> valueSpace, Type type) {
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side float value is less than or equal to the right hand side float value, else false.
+     */
+    public static boolean compareValueLessThanOrEqual(double lhsValue, double rhsValue) {
+        return compareValues(lhsValue, rhsValue) <= 0;
+    }
+
+    /**
+     * Check if left hand side float value is greater than the right hand side float value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side float value is greater than the right hand side float value, else false.
+     */
+    public static boolean compareValueGreaterThan(double lhsValue, double rhsValue) {
+        return compareValues(lhsValue, rhsValue) > 0;
+    }
+
+    /**
+     * Check if left hand side float value is greater than or equal to the right hand side float value.
+     *
+     * @param lhsValue The value on the left hand side
+     * @param rhsValue The value on the right hand side
+     * @return True if left hand side float value is greater than or equal to the right hand side float value, else
+     * false.
+     */
+    public static boolean compareValueGreaterThanOrEqual(double lhsValue, double rhsValue) {
+        return compareValues(lhsValue, rhsValue) >= 0;
+    }
+
+    public static int compareValues(Object lhsValue, Object rhsValue) {
+        return compareValues(lhsValue, rhsValue, "");
+    }
+
+    public static int compareValues(Object lhsValue, Object rhsValue, String direction) {
+        int lhsTypeTag = TypeChecker.getType(lhsValue).getTag();
+        int rhsTypeTag = TypeChecker.getType(rhsValue).getTag();
+        boolean inRelationalExpr = false;
+        if (direction.isEmpty()) {
+            inRelationalExpr = true;
+        }
+        boolean isAscending = direction.equals("ascending");
+        // Compare(x,()) is UN if x is not ().
+        // () should come last for CompareA(x, y) and CompareD(x, y).
+        if (lhsValue == null) {
+            if (rhsValue == null) {
+                return 0;
+            }
+            if (inRelationalExpr) {
+                throw ErrorUtils.createUnorderedTypesError(TYPE_NULL, rhsValue);
+            }
+            if (isAscending) {
+                return 1;
+            }
+            return -1;
+        }
+        if (rhsValue == null) {
+            if (inRelationalExpr) {
+                throw ErrorUtils.createUnorderedTypesError(lhsValue, TYPE_NULL);
+            }
+            if (isAscending) {
+                return -1;
+            }
+            return 1;
+        }
+        if (TypeTags.isStringTypeTag(lhsTypeTag) && TypeTags.isStringTypeTag(rhsTypeTag))  {
+            return codePointCompare(lhsValue.toString(), rhsValue.toString());
+        }
+        if (lhsTypeTag == TypeTags.BOOLEAN_TAG && rhsTypeTag == TypeTags.BOOLEAN_TAG)  {
```

Comment:
We can simple of code as below

if(lhsTypeTag == rhsTypeTag) {
swtch (lhsTypeTag) {
case BOOLEAN;
// bolean logic
case INT;
.... <other tags)
}

}